### DISCUSSION TITLE
Is there a way to load some text files from content folder?

### DISCUSSION THREAD

**Discussion by piti6** [nxrighthere/UnrealCLR#156] [Help] [answered]
Hello, I'm totally new in UE4 and UnrealCLR, (and really appreciated that I can use .Net on UE4 :joy:) 
I'm trying to load some text files like csv/jsons as I need to write easily editable game data. 
so far, as I can access contents folder by System.IO, (on editor at least) is it safe to load file by using System.IO? 
or is there alternative solution to do this? Like DataTable and AssetRegistryModule or something like that? 
I saw there is some methods that I can load some Blueprints or static meshes by path parameter.

**Comment by nxrighthere** [comment, marked as answer]
> so far, as I can access contents folder by System.IO, (on editor at least) is it safe to load file by using System.IO?

Yes, it's safe and this is pretty much a regular way to load content.

> or is there alternative solution to do this? Like DataTable and AssetRegistryModule or something like that?

Generally, [serialization of objects](https://docs.microsoft.com/en-us/dotnet/standard/serialization/system-text-json-how-to?pivots=dotnet-5-0) using JSON is preferable, but you can use any other data-interchange format you like.

> I saw there is some methods that I can load some Blueprints or static meshes by path parameter.

The framework of the plugin covers some functionality for loading engine-specific content from a drive, for custom content, you are free to use any available 3rd-party libraries.
